internal/client/blueprints.go

```diff
@@ -0,0 +1,167 @@
+// Package client - blueprints contains functions for the blueprint API
+// Copyright (C) 2020 by Red Hat, Inc.
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// PostTOMLBlueprintV0 sends a TOML blueprint string to the API
+// and returns an APIResponse
+func PostTOMLBlueprintV0(socket, blueprint string) *APIResponse {
+	body, err := PostTOML(socket, "/api/v0/blueprints/new", blueprint)
+	if err != nil {
+		return err
+	}
+	return NewAPIResponse(body)
+}
+
+// PostTOMLWorkspaceV0 sends a TOML blueprint string to the API
+// and returns an APIResponse
+func PostTOMLWorkspaceV0(socket, blueprint string) *APIResponse {
+	body, err := PostTOML(socket, "/api/v0/blueprints/workspace", blueprint)
+	if err != nil {
+		return err
+	}
+	return NewAPIResponse(body)
+}
+
+// PostJSONBlueprintV0 sends a JSON blueprint string to the API
+// and returns an APIResponse
+func PostJSONBlueprintV0(socket, blueprint string) *APIResponse {
+	body, err := PostJSON(socket, "/api/v0/blueprints/new", blueprint)
+	if err != nil {
+		return err
+	}
+	return NewAPIResponse(body)
+}
+
+// PostJSONWorkspaceV0 sends a JSON blueprint string to the API
+// and returns an APIResponse
+func PostJSONWorkspaceV0(socket, blueprint string) *APIResponse {
+	body, err := PostJSON(socket, "/api/v0/blueprints/workspace", blueprint)
+	if err != nil {
+		return err
+	}
+	return NewAPIResponse(body)
+}
+
+// DeleteBlueprintV0 deletes the named blueprint and returns an APIResponse
+func DeleteBlueprintV0(socket, bpName string) *APIResponse {
+	body, err := DeleteRaw(socket, "/api/v0/blueprints/delete/"+bpName)
+	if err != nil {
+		return err
+	}
+	return NewAPIResponse(body)
+}
+
+// DeleteWorkspaceV0 deletes the named blueprint's workspace and returns an APIResponse
+func DeleteWorkspaceV0(socket, bpName string) *APIResponse {
+	body, err := DeleteRaw(socket, "/api/v0/blueprints/workspace/"+bpName)
+	if err != nil {
+		return err
+	}
+	return NewAPIResponse(body)
+}
+
+// ListBlueprintsV0 returns a list of blueprint names
+func ListBlueprintsV0(socket string) ([]string, *APIResponse) {
+	body, err := GetJSONAll(socket, "/api/v0/blueprints/list")
+	if err != nil {
+		return nil, err
+	}
+	var resp BlueprintsListV0
+	jerr := json.Unmarshal(body, &resp)
```

Comment:
I think it would make sense to pass an `interface{}` type to `GetJSONAll()` and let that deal with the `Unmarshal()`, similarly to my suggestion above for letting the lower layers do the marshalling.

Reply:
I really don't like interface{}, it should be used sparingly. It's too easy to lean on it and lose type safety. We know what the response type should be at this point, so it should be enforced.

Reply:
My point was just to push the `json.Unmarshal()` into `GetJSONAll()`, which would encapsulate the JSON unmarshalling. The fact that the type is `interface{}` is just because that's what `json.Unmarshal()` takes.